packages/wallet-lib/src/plugins/Workers/TransactionSyncStreamWorker/methods/processChunks.js

```diff
@@ -1,121 +1,134 @@
-    // 1) The transaction itself needs to be imported
-    const { addressesGenerated: addressesGeneratedCount } = await self
-      .importTransactions(addressesTransactionsWithoutMetadata);
-    // 2) Transaction metadata need to be fetched and imported as well.
-    //    as such event might happen in the future
-    //    As we require height information, we fetch transaction using client
-
-    const awaitingMetadataPromises = addressesTransaction.transactions
-      .map((transaction) => self.handleTransactionFromStream(transaction)
-        .then(({
-          transactionResponse,
-          metadata,
-        }) => [transactionResponse.transaction, metadata]));
-
-    Promise
-      .all(awaitingMetadataPromises)
-      .then(async (transactionsWithMetadata) => {
-        // Import into account
-        const { mostRecentHeight } = await self.importTransactions(transactionsWithMetadata);
-
-        if (mostRecentHeight !== -1) {
-          this.setLastSyncedBlockHeight(mostRecentHeight, true);
-        }
-
-        // Schedule save state after all chain data has been imported
-        this.storage.scheduleStateSave();
-      })
-      .catch((err) => {
-        logger.error('Error while importing transactions', err);
-      });
-
-    self.hasReachedGapLimit = self.hasReachedGapLimit || addressesGeneratedCount > 0;
-
-    if (self.hasReachedGapLimit && self.stream) {
-      logger.silly('TransactionSyncStreamWorker - end stream - new addresses generated');
-
-      if (isBrowser()) {
-        // Under browser environment, grpc-web doesn't call error and end events
-        // so we call it by ourselves
-        await new Promise((resolveCancel) => setImmediate(() => {
-          self.stream.cancel();
-          const error = new GrpcError(GrpcErrorCodes.CANCELLED, 'Cancelled on client');
-
-          // call onError events
-          self.stream.f.forEach((func) => func(error));
-
-          // call onEnd events
-          self.stream.c.forEach((func) => func());
-          resolveCancel();
-        }));
-      } else {
-        // If there are some new addresses being imported
-        // to the storage, that mean that we hit the gap limit
-        // and we need to update the bloom filter with new addresses,
-        // i.e. we need to open another stream with a bloom filter
-        // that contains new addresses.
-
-        // DO not setting null this.stream allow to know we
-        // need to reset our stream (as we pass along the error)
-        // Wrapping `cancel` in `setImmediate` due to bug with double-free
-        // explained here (https://github.com/grpc/grpc-node/issues/1652)
-        // and here (https://github.com/nodejs/node/issues/38964)
-        await new Promise((resolveCancel) => setImmediate(() => {
-          self.stream.cancel();
-          resolveCancel();
-        }));
+      if (addressesGenerated > 0) {
+        this.reconnectOnNewBlock = true;
```

Comment:
We need to extend bloom filter right away and not wait for new block, since the gap limit is already closed at this point

Reply:
Done here https://github.com/dashpay/platform/blob/feat/integrate-with-block-headers-stream/packages/wallet-lib/src/plugins/Workers/TransactionsSyncWorker/TransactionsReader.js#L300